### DISCUSSION TITLE
Should PyPDF2 2.0.0 expose everything directly under PyPDF2?

### DISCUSSION THREAD

**Discussion by MartinThoma** [py-pdf/pypdf#869] [Ideas]
Currently, PyPDF2 has the following public modules:

* errors: All Exceptions
* generic: NullObject, BooleanObject, ArrayObject, DictionaryObject, IndirectObject, ...
* utils: readUntilRegex, skipOverComment, skipOverWhitespace, ...
* pdf: will be removed with PyPDF2 2.0.0
* merger: will become private to be consistant with `_reader.py` and `_writer.py`.
* pagerange: PageRange, parse_filename_page_ranges 

Is there any reason NOT to expose all public classes / functions under `PyPDF2` (via `__all__`) directly?

The advantage of exposing it there would be:

1. The users don't have to memorize how PyPDF2 is internally structured
2. PyPDF2 can be restructured easily as the public interface is clearly defined in one place

**Comment by pubpub-zz** [comment]
I agree about errors/generic/pagerange/merger
I have some doubts about utils :  unless I'm wrong they are only dealing with internal functions,
Also what could be a good invitation to use the new syntax could be to make public the new names (PdfReader) but not the old one (PdfFileReader)
